### PR TITLE
Fix an integer overflow issue

### PR DESCRIPTION
**Pull request by quangnh89**

### CODE REVIEW

File: contrib/windows_kernel/libc.cpp [hunk 1]
```diff
@@ -33,10 +33,19 @@ __cdecl malloc(
 	__in size_t size
 	)
 {
+    /*
+    A specially crafted size value can trigger the overflow.
+    If the sum in a value that overflows or underflows the capacity of the type,
+    the function returns nullptr.
+    */
+    size_t number_of_bytes = 0;
+    if (!NT_SUCCESS(RtlSizeTAdd(size, sizeof(MEMBLOCK), &number_of_bytes))){
```
Comment (aquynh) [inline contrib/windows_kernel/libc.cpp]: i think this file uses tabs for indentation, not spaces.

**Comment by aquynh** [comment]
looks good to me. @zer0mem, can you ack?

File: windows/winkernel_mm.c [hunk 2]
```diff
@@ -33,8 +34,16 @@ void * CAPSTONE_API cs_winkernel_malloc(size_t size)
 
 	// FP; a use of NonPagedPool is required for Windows 7 support
 #pragma prefast(suppress : 30030)		// Allocating executable POOL_TYPE memory
-	CS_WINKERNEL_MEMBLOCK *block = (CS_WINKERNEL_MEMBLOCK *)ExAllocatePoolWithTag(
-			NonPagedPool, size + sizeof(CS_WINKERNEL_MEMBLOCK), CS_WINKERNEL_POOL_TAG);
+	size_t number_of_bytes = 0;
+	CS_WINKERNEL_MEMBLOCK *block = NULL;
```
Comment (aquynh) [inline windows/winkernel_mm.c]: why do you have to initialize these variables here? 

same question for number_of_bytes in contrib/windows_kernel/libc.cpp

Comment (quangnh89) [inline windows/winkernel_mm.c]: It's just my habit. I initialize every variables when i declare them. In some cases, it is unnecessary. In the other cases, it prevents attacks on uninitialized local variables. I don't think this is a big problem. Compiler will optimize everything.

**Comment by aquynh** [comment]
merged, thanks!

**Pull request merged**